src/main/java/com/eternalcode/combat/fight/pearl/FightPearlController.java

```diff
@@ -73,4 +77,47 @@ void onInteract(PlayerInteractEvent event) {
 
         this.fightPearlManager.markDelay(uniqueId);
     }
+
+    @EventHandler(priority = EventPriority.LOW)
+    void onEntityDamage(EntityDamageByEntityEvent event) {
+        if (!this.settings.pearlThrowControlEnabled) {
+            return;
+        }
+
+        if (!(event.getEntity() instanceof Player player)) {
+            return;
+        }
+
+        UUID playerUniqueId = player.getUniqueId();
+
+        if (!(event.getDamager() instanceof EnderPearl)) {
+            return;
+        }
+
+        if (event.getCause() != EntityDamageEvent.DamageCause.FALL) {
+            return;
+        }
+
+        if (!this.settings.pearlThrowMarksCombat) {
+            event.setCancelled(true);
+        }
+
+        if (this.settings.pearlThrowDamageEnabled) {
+            this.damagePlayerIfEventIsCancelled(event, player);
+        }
+        else {
+            if (this.settings.pearlThrowDamageEnabledInCombat && this.fightManager.isInCombat(playerUniqueId)) {
+                this.damagePlayerIfEventIsCancelled(event, player);
+            }
+            else {
+                event.setDamage(0.0D);
```

Comment:
Jeśli chodzi o tego feature to, można go zrobić w innym listenerze który ignoruje anulowane eventy i słucha na HIGHEST.
Pozwoli to na dobry flow zgrany z takimi pluginami od regionów itd. 

De fakto będzie to polegać na setDamage(0.0D) bez konieczności symulowania obrażeń